agent_api_rest/src/issuance/credential_issuer/token.rs

```diff
@@ -100,19 +104,20 @@ pub mod tests {
         assert_eq!(token_response.token_type, "bearer");
         assert!(token_response.c_nonce.is_some());
 
-        token_response.access_token
+        Some(token_response.access_token)
     }
 
+    #[serial_test::serial]
     #[tokio::test]
     async fn test_token_endpoint() {
         let issuance_state = in_memory::issuance_state(Service::default(), Default::default()).await;
         initialize(&issuance_state, startup_commands(BASE_URL.clone())).await;
-
         let mut app = router(issuance_state);
 
         credentials(&mut app).await;
-        let pre_authorized_code = offers(&mut app).await;
+        let pre_authorized_code: String = offers(&mut app).await.unwrap();
 
-        let _access_token = token(&mut app, pre_authorized_code).await;
+        let _access_token = token(&mut app, Some(pre_authorized_code)).await;
+        assert!(_access_token.is_some());
```

Comment:
This won't be necessary since the `token` function should always return a `String` (instead of `Option<String>`)

Reply:
So related to the comment above, this line:
```rust
let _access_token = token(&mut app, Some(pre_authorized_code)).await;
```

Should be changed to:
```rust
let _access_token = token(&mut app, pre_authorized_code).await;
```

because the `token` function should expect `pre_authorized_code: String` instead of `pre_authorized_code: Option<String>` :)